### PR TITLE
Add a tf.contrib.image.translate function

### PR DESCRIPTION
**Pull request by ringw**
Opening as a replacement for #10748. Made a small simplification. @sampepose

### CODE REVIEW

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (vrv) [inline tensorflow/contrib/image/python/ops/image_ops.py]: does this need to be public, or can it be an hidden?

Comment (ringw) [inline tensorflow/contrib/image/python/ops/image_ops.py]: We already expose angles_to_projective_transforms so that users can compose transformation matrices, so my plan was to make similar public methods for any transformation.

**Review by vrv — COMMENTED** [review]
Letting eugene look at the rest of this.

**Comment by sampepose** [comment]
Thanks, @ringw! Sorry I haven't gotten a chance to finish off that PR. Lgtm

**Comment by ahundt** [comment]
@sampepose I think you need to sign the CLA before this can be merged. Otherwise it will be stuck forever. :-)

**Comment by sampepose** [comment]
hmm I've signed the CLA, not sure why this isn't going through.

SP

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: name= arg on any function that creates new ops., use it inside a "with name_scope(...):" before creating any ops.  here, elsewhere.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: what if .get_shape() returns TensorShape(None)?

**Comment by ringw** [comment]
@ahundt @sampepose The googlebot comment says "The good news is that everyone that needs to sign a CLA (the pull request submitter and all commit authors) have done so. Everything is all good there.", so hopefully the CLA is OK?

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: That is an issue throughout, but it's really useful to do this validation eagerly and be able to return a tensor with the correct shape (HW, HWC, or NHWC). Added docstrings indicating which arguments need a static shape with a known rank.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Added name= args and tf.name_scope(name) throughout.

**Comment by ahundt** [comment]
Oh I see you're right, I had only looked at the `cla: no` label.

**Comment by drpngx** [comment]
@ringw we really need that CLA. Could you double check? Usually, it's a problem with the email address, or commits getting co-mingled the PR. Thanks.

**Comment by ringw** [comment]
@drpngx The only details I see about the CLA are: "The bad news is that it appears that one or more commits were authored by someone other than the pull request submitter. We need to confirm that they're okay with their commits being contributed to this project. Please have them confirm that here in the pull request."

@sampepose confirmed in this pull request that he's OK with contributing his commit. Are there any further steps necessary? Sam's commit has sampepose@gmail.com which appears to be his email on his profile, and my commits have ringwalt@google.com. I don't see any other commits that got co-mingled.

**Comment by drpngx** [comment]
@sampepose Please leave a comment that you're OK with it, and then we can override and merge.

However, we need an OK from @ebrevdo first. Thanks.

**Comment by ringw** [comment]
Sam already replied with "LGTM".

**Comment by drpngx** [comment]
@ringw I saw that, I just want a clear comment.

**Comment by sampepose** [comment]
Looks good to me. Merge away.

**Comment by sb2nov** [comment]
@drpngx is this okay to merge ?

**Comment by drpngx** [comment]
I think we'd still like an approval from @ebrevdo. Once that is done, @gunan, @martinwicke or @jhseu can override the CLA flag and merge.

File: tensorflow/contrib/image/python/kernel_tests/image_ops_test.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/kernel_tests/image_ops_test.py]: Move the new session inside the loop.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: name=None and use the default second argument to name_scope below

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: Check that shape.ndims is not none first

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: You mean a matrix?

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: name, "translate"):

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: Check ndims first

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ebrevdo) [inline tensorflow/contrib/image/python/ops/image_ops.py]: Explain what you're doing here. Match dtype to image dtype?

**Review by ebrevdo — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/image/python/kernel_tests/image_ops_test.py
Comment (ringw) [reply tensorflow/contrib/image/python/kernel_tests/image_ops_test.py]: Done.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Done throughout.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Done throughout.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Done.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Done throughout.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Done.

File: tensorflow/contrib/image/python/ops/image_ops.py
Comment (ringw) [reply tensorflow/contrib/image/python/ops/image_ops.py]: Actually, this just processes the translation matrices, which are always float32. Added a comment to clarify this, and describe the translation matrix.

**Comment by ringw** [comment]
Ping @ebrevdo

**Review by ebrevdo — APPROVED** [review]

**Comment by ringw** [comment]
@drpngx @gunan Could we go ahead and override the CLA flag and merge? Thanks!

**Comment by gunan** [comment]
Sorry for the delay, merged.

**Comment by ringw** [comment]
No problem!

**Pull request merged**
